File: backend/src/main/java/com/oficina/nikima/model/Moto.java
package com.oficina.nikima.model;

import lombok.*;
import java.time.LocalDateTime;
import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Moto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String modelo;
    private String placa;
    private LocalDateTime dataEntrada;

    @Enumerated(EnumType.STRING)
    private Status status;

    private String mecanicoResponsavel;
    private String funcionarioNota;

    @OneToMany(mappedBy = "moto", cascade = CascadeType.ALL)
    private List<Servico> servicos;
}

File: frontend/src/app/models/moto.ts
import { Status } from './status';
import { Servico } from './servico';

export interface Moto {
  id?: number;
  modelo: string;
  placa: string;
  dataEntrada?: string;
  status: Status;
  mecanicoResponsavel: string;
  funcionarioNota: string;
  servicos: Servico[];
}
